### ISSUE TITLE
Event fired from mounted hook; how to listen?

### ISSUE THREAD

**Issue by ale-grosselle** [vuejs/vue-web-component-wrapper#41]
Hi,
I have a little problem; I'am trying to listen an event fired from mouted hook, but seem not to be possible, because mounted function are always fired before the event callbacks definition

**Comment by ale-grosselle** [comment]
Hi @karol-f 
https://codepen.io/ale-grosselle/pen/oQQMRe
I understood the problem: is the code flow.
If you add before: 
1) wrapped component  definition
2) html component
3) event registration
Never fired ready

Instead:
1) html component
2) event registration
3) wrapped component definition

Is not easy to solve, because I have my component already registered and I lost all events fired from created.

**Comment by karol-f** [comment]
But that's how it's work - if you register Custom Elements browser will immediately start consuming its callbacks (e.g. init Vue component).
That's why you need to addListener before registration (in most cases it will be just reordering js scripts) or register listener in. e.g. createdCallback of Custom Element.

**Comment by ale-grosselle** [comment]
Yes you have reason, but after I define the web component using wrap function, I lost all "mounted" hook.

I Have implemeted a WA in my fork:
https://github.com/ale-grosselle/vue-web-component-wrapper/commit/81b16f97efcf5661a9516808cb625be97ced53f8
``
window.setTimeout(
        this.shadowRoot.appendChild(wrapper.$el)	          function () {
            wrapper.$mount()
            if (useShadowDOM) this.shadowRoot.appendChild(wrapper.$el)
            else this.appendChild(wrapper.$el)
          }.bind(this), 0
        )
``

**Comment by i7eo** [comment]
> wrapper

u can try `wrapper.__isMounted`
